main.py

```diff
@@ -558,6 +562,21 @@ def main(cl_arguments):
     if args.do_full_eval:
         log.info("Evaluating...")
         splits_to_write = evaluate.parse_write_preds_arg(args.write_preds)
+        diagnostic_tasks = []
+        target_tasks_to_remove = []
+        for index, task in enumerate(target_tasks):
+            if task.name in task_modules.ALL_DIAGNOSTICS:
+                diagnostic_tasks.append(task)
+                target_tasks_to_remove.append(index)
+        # To avoid evaluating twice on a diagnostic task.
+        for index in sorted(target_tasks_to_remove, reverse=True):
+            del target_tasks[index]
+        for task in diagnostic_tasks:
+            ckpt_path = get_best_checkpoint_path(args, "eval", task.name)
```

Comment:
Where do we substitute task.name for the appropriate trained model? For SuperGLUE diagnostics, we want task.name to be rte-superglue, not the name of the diagnostic.

Reply:
Right now, the only way to run diagnostics is to have a model in pretraining stage, and then superglue-diagnostics in evaluation. We don't need to switch the task.name to be rte-superglue since it's in pretraining stage and the checkpoints for pretraining stage do not depend on task name. 
Ideally, what we would want is to separate target_tasks from eval_tasks entirely so that it would be possible to pretrain, then target train, then use that target train to evaluate on diagnostics, but that will entail a bigger rewrite.

Reply:
I think at some point, you'll have to maintain a list of pretraining tasks, target tasks that you train on, target tasks that you don't train on. We probably want to maintain only giving users the choice between pretraining and target tasks (merging the second two categories into one) and handling the task management ourselves, like we're doing here.